generators/generator-base-entity.cjs

```diff
@@ -63,131 +114,230 @@ class JHipsterBaseEntityGenerator extends BaseBlueprintGenerator {
+    return this.getExistingEntityNames().map(entityName => ({ entityName, entityStorage: this.getEntityConfig(entityName, true) }));
+  }
+
+  /**
+   * @private
+   * Get entities to prepare.
    * @returns {object[]}
    */
   getEntitiesDataToPrepare() {
-    return this.sharedData.getEntities(this.getExistingEntityNames());
+    return this.sharedData.getEntities().map(({ entityName, ...data }) => ({
+      description: entityName,
+      entityName,
+      ...data,
+    }));
   }
 
   /**
-   * Get entities names to generate.
+   * @private
+   * Get entities and fields to prepare.
+   * @returns {object[]}
+   */
+  getEntitiesFieldsDataToPrepare() {
+    return this.getEntitiesDataToPrepare()
+      .map(({ entity, entityName, ...data }) => {
+        if (!entity.fields) return [];
+
+        return entity.fields.map(field => ({
+          entity,
+          entityName,
+          ...data,
+          field,
+          fieldName: field.fieldName,
+          description: `${entityName}#${field.fieldName}`,
+        }));
+      })
+      .flat();
+  }
+
+  /**
+   * @private
+   * Get entities and relationships to prepare.
+   * @returns {object[]}
+   */
+  getEntitiesRelationshipsDataToPrepare() {
+    return this.getEntitiesDataToPrepare()
+      .map(({ entity, entityName, ...data }) => {
+        if (!entity.relationships) return [];
+
+        return entity.relationships.map(relationship => ({
+          entity,
+          entityName,
+          ...data,
+          relationship,
+          relationshipName: relationship.relationshipName,
+          description: `${entityName}#${relationship.relationshipName}`,
+        }));
+      })
+      .flat();
+  }
+
+  /**
+   * @private
+   * Get entities to post prepare.
+   * @returns {object[]}
+   */
+  getEntitiesDataToPostPrepare() {
+    return this.getEntitiesDataToPrepare();
+  }
+
+  /**
+   * @private
+   * Get entities to write.
    * @returns {object[]}
    */
   getEntitiesDataToWrite() {
-    return this.sharedData.getEntities(this.getExistingEntityNames());
+    const { entities = [] } = this.options;
+    let entitiesDefinitions = this.sharedData.getEntities();
+    if (entities.length > 0) {
+      entitiesDefinitions = entitiesDefinitions.filter(({ entityName }) => entities.includes(entityName));
+    }
+    return { entities: entitiesDefinitions.map(({ entity }) => entity) };
   }
 
-  get [QUEUE_EACH_ENTITY_PRIORITY]() {
-    return {
-      queueConfiguringEachEntity() {
-        this.debug(`Queueing entity tasks ${CONFIGURING_EACH_ENTITY_PRIORITY}`);
-        const tasks = this.extractTasksFromPriority(CONFIGURING_EACH_ENTITY_PRIORITY, { skip: false });
-        this.getEntitiesToConfigure().forEach(({ entityName, entityStorage, entityConfig }) => {
-          this.debug(`Queueing entity tasks ${CONFIGURING_EACH_ENTITY_PRIORITY} for ${entityName}`);
+  /**
+   * @private
+   * Queue entity tasks.
+   */
+  queueEntityTasks() {
+    this.queueTask({
```

Comment:
Maybe one sub-method by step?

Reply:
I prefer it together. It's easier to see the end result.